common/client.go

```diff
@@ -0,0 +1,338 @@
+// Copyright (c) 2015 Pani Networks
+// All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License. You may obtain
+// a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations
+// under the License.
+
+package common
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/pborman/uuid"
+	"io/ioutil"
+	"log"
+	"math"
+	"net/http"
+	"net/url"
+	"reflect"
+	"time"
+)
+
+// Rest Client for the Romana services. Incorporates facilities to deal with
+// various REST requests. 
+type RestClient struct {
+	url    *url.URL
+	client *http.Client
+	config *RestClientConfig
+}
+
+type RestClientConfig struct {
+	TimeoutMillis int64
+	Retries       int
+}
+
+func GetDefaultRestClientConfig() RestClientConfig {
+	return RestClientConfig{TimeoutMillis: DefaultRestTimeout, Retries: DefaultRestRetries}
```

Comment:
Why don't we have the sanity checking for the values here in this function? Currently, every time NewRestClient() is called those config values are evaluated, in case of invalid values there is a log message and we default to something instead. Could we not do this just once here in this function? That way, we keep the data clean 'at the source', so to speak, and don't have to run through those tests all the time.